docs/how-to/test-update-system.md

```diff
@@ -0,0 +1,281 @@
+     1. `k: 9`
+     2. `genesisSlotDurationSec: 15`
+     3. `networkDiameter: 10`
+3. Generate `genesis.bin` for US testing cluster:
+
+   (!) Ensure you compiled branch `csl-xx` in prod mode, e.g. with `./util-scripts/build.sh --prod`
+
+   ```
+M=4 N=200 ./util-scripts/generate-genesis.sh
+cp genesis-qanet-<current_date>/genesis.* .
+   ```
+4. Commit changes:
+```
+git add genesis.info genesis.bin core/constants*prod.yaml
+git commit -m "Updated constants, genesis"
+git push --set-upstream origin csl-xx
+```
+
+5. Build `csl-xx` branch in `--prod` mode
+
+   This is needed for later steps.
+
+## Cluster for US testing, installers
+
+You need launch cluster with name `csl-xx` on `staging` user of cardano-deployer. This process is described in different place (**insert_link_here**).
+
+Use HEAD of `csl-xx` branch as commit for `cardano-sl`.
+
+After launching deployment, you will have:
+* `system-start` value
+* `ip-dht-mappings` file
+
+After build of `csl-xx` branch is finished in CI, you need to update `daedalus` for installers to be built:
+1. Use same branch `cardano-sl-Y.Z` of daedalus, branch out `csl-xx`
+2. Change branch from `cardano-sl-Y.Z` to `csl-xx` in `.travis.yml`, `appveyor.yaml`
+3. Update `system-start` in `installers/Launcher.hs`
+4. Replace `ip-dht-mappings` in `installers/data/`
+5. Commit & push changes
+
+Then wait for CI to build installers for base version.
+
+## Test base version
+
+1. Download installers for `csl-xx`
+2. Install & launch daedalus
+3. Check:
+    * blocks are retrieved
+    * transactions are sent
+
+## Prepare updated version
+
+1. Update `constants.yaml`:
+    * increment `lastKnownBVMinor`
+    * increment `applicationVersion`
+2. Commit changes
+3. Wait for `cardano-sl` to be built
+4. Trigger `daedalus` build (installers)
+
+## Propose update & vote for it
+
+### Prepare node keys
+
+1. Take genesis deployed to `csl-xx` cluster
+2. ```
+cp -R <genesis_directory>/nodes nodes2
+cd nodes2
+for i in {0..4}; do stack exec -- cardano-keygen --rearrange-mask key$i.sk; done
+```
+
+### Cardano-wallet
+
+Launch `cardano-wallet` with CLI:
+
+```
+stack exec -- cardano-wallet --system-start 1498572075 --log-config log-config-prod.yaml --logs-prefix "logs/abc4" --db-path db-abc4 --peer 52.57.33.126:3000 --peer 52.57.215.206:3000 --peer 34.253.40.70:3000 --peer 52.211.96.220:3000 repl
+```
+
+(!) Replace:
+
+1. `--peer` parameters with ips of the `csl-xx` cluster nodes
+2. `--system-start` parameter with system start for `csl-xx` cluster
+
+Then you will appear in `repl` mode and would need to perform few actions:
+
+#### Import keys
+
+```
+add-key nodes2/key1.sk
+add-key nodes2/key2.sk
+add-key nodes2/key3.sk
+add-key nodes2/key4.sk
+```
+
+Then if you execute
+
+```
+listaddr
+```
+
+You should see something like:
+
+```
+> listaddr
+Available addresses:
+    #0:   1feg76nzCcJrhwX4Z9NXc9iwNwo1UxUdkaF3XwdyFhtDdst (PK: eMrhrnbwqmRmeHNQW6kyrD9gs4sMtUyGRqPuLsiZaaTLhHY4uojgp7imLNX6tsECechdqLUB92APfs6Er1BJv6E)
```

Comment:
The output of `listaddr` is more verbose after https://github.com/input-output-hk/cardano-sl/pull/1453, the user will see something like this instead:

```
Available addresses:
    #0:   addr:      9d5cGKg5x3Vu3g2APH2QJMjt8WyRKfSE4Nohbz1q9y5FZs51mMWkFCjSt7ps8TwJeSES55jaXhyV3wVJr69Q8CLs2cu4xHpDDx5N
          pk base58: bNfckjryxQLmVh617K6HVPpeKPJWjZsGkNGsfgm64mQQN4Q4sShp73NRDJVXA7pM1Eqt5tcHvTsHsg7kAT7R6bMPix
          pk hex:    2c63dd8d53e64913131b4b8adf7930aff642828f734d64c06bc775863a868c871e9d34286e5aad95515d6cb3a40dd29a3a70ceb345cba6ad6687236be4c922f9
          pk hash:   6dab12ab37cc34b40fc6188969699159c78a9231901b166a897162eb
    #1:   addr:      9d5cHXm3XqMPtyCwKyLEBdqK6bunok466ogy13UbpSggbqaRjkr36n2JfrY3RFvDyT5Qt3Hz6XXgLDepTZVRqDybCZQcJFd2rosi
          pk base58: bNhQZ5x9DGYeV4gLyk1Axa4yp4xzrzZBskXov4Vk7AkrogndKgpkiMLJrNqaCJQAJmyrzAbeq4WtqkYH7mBisM2e2w
          pk hex:    85e7090aafc05f65e7e3774fef2ab12b4c0541f16436fded33472efe41b8c60cc4bce3b4728221f1ea5878777109ca3198a22f9932482eac09bf533e22c0388c
          pk hash:   f4a8b9b0f4671cb2486432619f55309ef4a424a8b3ab191ed7484037
    #2:   addr:      9d5cAS1VZHwNVkdBLVvNPmY9mpa4XuMPrn7DsQe7tmWjAqMDYMYzw54kLfMztQfJPVQN3uJnBCmkUD5f4YKogWXEeeXiBXTY1XqK
          pk base58: bNjEfqa4agzaXii4oEtugN3mMA6UykrfdASYFwNZsJLjoVnhr5FWwyg5ZLHZAqR6XXygPtdbu1bbtJhx1ptddPLWFp
          pk hex:    e169cabc73c70664b4f679a7910941da55ca1736a67513b98bf430c8e3a4490cdb3d70c322fa1943a4e78f90d6207dd6a90a35dd1ccf8dc0d333684d18724ec7
          pk hash:   6056d5a781311a76d5cd7b4f43823de687bb0cba5c4482bf4579fc9d
    #3:   addr:      9d5cKX74agz6Dcu3ZTibzWtR11aciNkSQKh4mFFQMEEdvvbvDwbNgMFK3ZW7Pxx2BERGSNX7G4FUXbYsSZuGNdp8ZCa1EJtQnRYj
          pk base58: bNiDRg1JczbotuRLXEMmsPgpnY5cCE2Dt3oKJVxjp6kgUGgqCgpjctYPJJ3fD3641Zuj9bwJ3zAaMaJkVBGmKFjPn7
          pk hex:    ae52c28bdfcda269ce747b12389bbebb54872f83a81f8b9a2ed69134a9df75a8b8e46ebcfda5562cac6dadd839f132d4f53b6b332053868b296bd740a9872880
          pk hash:   a50d738df567fc3ec39c2e990cd8a95930ec3ccae0b98b2bbaace676
```

Reply:
(Probably don't copy the output from my comment, it's with different keys)